azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -323,6 +326,43 @@ private void addNetworkSecurityGroup() {
 
    }
 
+   // Helper function for simplifying an array of ports to a list of ranges FirewallOptions expects.
+   public static List<String> simplifyPorts(int[] ports){
+      if ((ports == null) || (ports.length == 0)) {
+         return null;
```

Comment:
This should never happen at this point, as it is already validated before. Consider it a precondition of this method and better throw an `IllegalArgumentException`. Otherwise you must add a check in the caller to verify that the returned value is not null, to avoid NPEs.